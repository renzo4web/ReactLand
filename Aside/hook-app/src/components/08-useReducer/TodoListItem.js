const TodoListItem = ({
  k,
  onRemove,
  onCompleted,
  item,
}) => {
  const { todo, id, completed } = item;

  return (
    <li
      className={
        completed
          ? "bg-success list-group-item d-flex justify-content-between align-items-start "
          : "list-group-item d-flex justify-content-between align-items-start"
      }
      key={k}
    >
      <div className="ms-2 me-auto">
        <div className="fw-bold">{todo}</div>
        completed: {completed.toString()}
      </div>

      <button
        className="btn badge bg-danger rounded-pill"
        onClick={() => onRemove(todo)}
      >
        X
      </button>
      <button
        className="btn badge bg-success rounded-pill"
        onClick={() => onCompleted(id)}
      >
        V
      </button>
    </li>
  );
};
export default TodoListItem;
