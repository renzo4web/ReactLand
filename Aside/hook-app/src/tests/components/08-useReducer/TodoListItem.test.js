import { shallow } from "enzyme";
import React from "react";
import TodoListItem from "../../../components/08-useReducer/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";

describe("<TodoListItem /> tests", () => {
  const onRemove = jest.fn();
  const onCompleted = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      item={demoTodos[0]}
      onRemove={onRemove}
      onCompleted={onCompleted}
      k="44523"
    />
  );

  test("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should call the fn onRemove", () => {
    wrapper.find(".btn.badge.bg-danger").simulate("click");

    expect(onRemove).toHaveBeenCalled();
    expect(onRemove).toHaveBeenCalledWith(
      demoTodos[0].todo
    );
  });

  test("should call the fn onCompleted", () => {
    wrapper.find(".btn.badge.bg-success").simulate("click");

    expect(onCompleted).toHaveBeenCalled();
    expect(onCompleted).toHaveBeenCalledWith(
      demoTodos[0].id
    );
  });

  test("should render the text showing the todo", () => {
    expect(wrapper.find(".fw-bold").text().trim()).toBe(
      demoTodos[0].todo
    );
  });
});
