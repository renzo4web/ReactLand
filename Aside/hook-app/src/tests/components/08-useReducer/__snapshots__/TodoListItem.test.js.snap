// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListItem /> tests should render properly 1`] = `
<li
  className="list-group-item d-flex justify-content-between align-items-start"
  key="44523"
>
  <div
    className="ms-2 me-auto"
  >
    <div
      className="fw-bold"
    >
      Learn React
    </div>
    completed: 
    false
  </div>
  <button
    className="btn badge bg-danger rounded-pill"
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="btn badge bg-success rounded-pill"
    onClick={[Function]}
  >
    V
  </button>
</li>
`;
